pkg/kubectl/cmd/util/diff/diffoptions.go

```diff
@@ -0,0 +1,190 @@
+		}
+	}
+
+	mapper, typer, err := f.UnstructuredObject()
+	if err != nil {
+		return err
+	}
+
+	o.Namespace, o.EnforceNamespace, err = f.DefaultNamespace()
+	if err != nil {
+		return err
+	}
+	o.Codec = runtime.NewCodec(f.JSONEncoder(), f.Decoder(true))
+	o.ext = "." + o.Output
+	o.Mapper = mapper
+	o.Typer = typer
+
+	r := resource.NewBuilder(o.Mapper, f.CategoryExpander(), o.Typer, resource.ClientMapperFunc(f.UnstructuredClientForMapping), unstructured.UnstructuredJSONScheme).
+		NamespaceParam(o.Namespace).DefaultNamespace().
+		FilenameParam(o.EnforceNamespace, &o.FilenameOptions).
+		Flatten().
+		Do()
+	err = r.Err()
+	if err != nil {
+		return err
+	}
+
+	err = r.Visit(func(info *resource.Info, err error) error {
+		if err != nil {
+			return err
+		}
+
+		o.DiffBuffer1, err = runtime.Encode(o.Codec, info.VersionedObject)
+		if err != nil {
+			return err
+		}
+
+		// Verify the object exists in the cluster
+		if err := info.Get(); err != nil {
+			if apierrors.IsNotFound(err) {
+				return err
+			} else {
+				return cmdutil.AddSourceToErr(fmt.Sprintf("retrieving current configuration of:\n%v\nfrom server for:", info), info.Source, err)
+			}
+		}
+
+		o.DiffBuffer2, err = kubectl.GetOriginalConfiguration(info.Mapping, info.Object)
+		if err != nil {
+			return cmdutil.AddSourceToErr(fmt.Sprintf("retrieving current configuration of:\n%v\nfrom server for:", info), info.Source, err)
+		}
+		if o.DiffBuffer2 == nil {
+			return cmdutil.UsageError(cmd, "no last-applied-configuration annotation found on resource: %s, to create the annotation, run the command with --create-annotation", info.Name)
+		}
+
+		//decode and delete the annotations fields
+		targetObj, err := runtime.Decode(unstructured.UnstructuredJSONScheme, o.DiffBuffer2)
+		if err != nil {
+			return nil
+		}
+		delete(targetObj.(runtime.Unstructured).UnstructuredContent()["metadata"].(map[string]interface{}), "annotations")
+
+		o.DiffBuffer2, err = runtime.Encode(o.Codec, targetObj)
+		if err != nil {
+			return err
+		}
+
+		return nil
+	})
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (o *DiffOptions) Validate(f cmdutil.Factory, cmd *cobra.Command) error {
+	return nil
+}
+
+func (o *DiffOptions) RunDiffLastApplied(f cmdutil.Factory, cmd *cobra.Command) error {
+	var err error
+	o.DiffBuffer1, err = FormatConvert(o.DiffBuffer1, o.Output)
+	if err != nil {
+		return err
+	}
+	o.DiffBuffer2, err = FormatConvert(o.DiffBuffer2, o.Output)
+	if err != nil {
+		return err
+	}
+	diff := cmdutil.NewDefaultCmdTool("diff", f.DiffEnvs())
+
+	_, _, err = diff.LaunchTempFile(fmt.Sprintf("%s-diff-", filepath.Base(os.Args[0])), o.ext, bytes.NewReader(o.DiffBuffer1), bytes.NewReader(o.DiffBuffer2))
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func FormatConvert(input []byte, o string) ([]byte, error) {
+	switch o {
+	case "json":
+		jsonBuffer := &bytes.Buffer{}
+		err := json.Indent(jsonBuffer, []byte(input), "", "  ")
```

Comment:
maybe, but I think the input parameter should be raw byte[] instead of indented data, any ideas on this ? comment this function like please do not misuse ? or we do some check in this function to make sure that won't happen ?